exporter/awsemfexporter/metric_translator.go

```diff
@@ -162,15 +162,17 @@ func translateGroupedMetricToCWMetric(groupedMetric *groupedMetric, config *Conf
 	}
 
 	var cWMeasurements []cWMeasurement
-	if len(config.MetricDeclarations) == 0 {
-		// If there are no metric declarations defined, translate grouped metric
-		// into the corresponding CW Measurement
-		cwm := groupedMetricToCWMeasurement(groupedMetric, config)
-		cWMeasurements = []cWMeasurement{cwm}
-	} else {
-		// If metric declarations are defined, filter grouped metric's metrics using
-		// metric declarations and translate into the corresponding list of CW Measurements
-		cWMeasurements = groupedMetricToCWMeasurementsWithFilters(groupedMetric, config)
+	if !config.DisableMetricExtraction { // If metric extraction is disabled, there is no need to compute & set the measurements
```

Comment:
We could have done this check during the time of translating a CWMetric into an EMF metric (json) but I decided to do this early on since there is no point to computing these measurements if they arent going to be used eventually - so saves on some compute.